File: Examples/FontKerning.cs

using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements;
using ceTe.DynamicPDF.Text;

namespace DynamicPDFCoreSuite.Examples
{
    class FontKerning
    {
        public static void Run()
        {
            Document document = new Document();
            Page page = new Page();
            string text = "To and WA are examples of when kerning would be used.";
            TextArea textArea = new TextArea(text, 10, 10, 400, 700, Font.TimesRoman);

            textArea.KerningEnabled = true;
            KerningValues kernValues = textArea.GetKerningValues();

            for (int i = 0; i < kernValues.Spacing.Length; i++)
            {
                kernValues.Spacing[i] = (short)(kernValues.Spacing[i] - 400);
            }

            page.Elements.Add(textArea);
            document.Pages.Add(page);
            document.Draw(Util.GetPath("Output/fontkerning-output.pdf"));
        }

    }
}

File: dynamicpdf-vbnet-examples/Examples/FontKerning.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements
Imports ceTe.DynamicPDF.Text

Namespace DynamicPDFCoreSuite.Examples
    Class FontKerning
        Public Shared Sub Run()
            Dim document As New Document()
            Dim page As New Page()
            Dim text As String = "To and WA are examples of when kerning would be used."
            Dim textArea As New TextArea(text, 10, 10, 400, 700, Font.TimesRoman)

            textArea.KerningEnabled = True
            Dim kernValues As KerningValues = textArea.GetKerningValues()

            For i As Integer = 0 To kernValues.Spacing.Length - 1
                kernValues.Spacing(i) = CType(kernValues.Spacing(i) - 400, Short)
            Next

            page.Elements.Add(textArea)
            document.Pages.Add(page)
            document.Draw(Util.GetPath("Output/fontkerning-output.pdf"))
        End Sub
    End Class
End Namespace

